crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2209,8 +2209,25 @@ impl<'db> TypeInferenceBuilder<'db> {
         } = unary;
 
         match (op, self.infer_expression(operand)) {
+            (UnaryOp::UAdd, Type::IntLiteral(value)) => Type::IntLiteral(value),
             (UnaryOp::USub, Type::IntLiteral(value)) => Type::IntLiteral(-value),
+            (UnaryOp::Invert, Type::IntLiteral(value)) => Type::IntLiteral(-(value + 1)),
```

Comment:
Can't we use Rust's bitwise logical `not` operator here, to mirror Python's bitwise logical `not` operator?

```suggestion
            (UnaryOp::Invert, Type::IntLiteral(value)) => Type::IntLiteral(!value),
```